app/src/main/java/com/cp/tasky/agenda/overview/data/remote/RemoteDataSource.kt

```diff
@@ -0,0 +1,30 @@
+package com.cp.tasky.agenda.overview.data.remote
+
+import com.cp.tasky.agenda.overview.data.model.OverviewApiResponse
+import com.cp.tasky.agenda.overview.data.model.OverviewSyncRequest
+import retrofit2.Response
+import javax.inject.Inject
+
+interface RemoteDataResource {
+    suspend fun getAgendaByDate(timeStamp: Long): Response<OverviewApiResponse>
+
+    suspend fun syncAgenda(agendaSyncRequest: OverviewSyncRequest)
+
+    suspend fun getFullAgenda(): Response<OverviewApiResponse>
+}
+
+class RemoteDataSourceImpl @Inject constructor(
+    private val overviewApi: OverviewApi
+): RemoteDataResource {
+    override suspend fun getAgendaByDate(timeStamp: Long): Response<OverviewApiResponse> {
+        return overviewApi.getAgendaByDate(timeStamp)
+    }
+
+    override suspend fun syncAgenda(agendaSyncRequest: OverviewSyncRequest) {
+        overviewApi.syncAgenda(agendaSyncRequest)
+    }
+
+    override suspend fun getFullAgenda(): Response<OverviewApiResponse> {
+        return overviewApi.getFullAgenda()
+    }
+}
```

Comment:
This is pretty much just a duplication of the Retrofit interface. Which problem does this abstraction and implementation solve? It has no real purpose other than delegating function calls

Reply:
Separation of Concerns:  Separating the data source interface from its implementation isolates the network communication details from the rest of your application. This adheres to the principle of separation of concerns, ensuring that each class or component has a single responsibility.

Testability: Interfaces make it easier to mock dependencies in unit tests. This allows you to test the repository logic independently of the actual network calls.

Flexibility and Extensibility: If the way you fetch data changes (e.g., switching from Retrofit to another networking library, or adding caching layers), you can do so without changing the repository’s code. Only the implementation of RemoteDataResource needs to change.

Handle network error .